### PR TITLE
Fix completion start with dollar

### PR DESCRIPTION
**Pull request by ukyo**
String disappears when I select a completion start with dollar.

### CODE REVIEW

File: TypeScript.py [hunk 1]
```diff
@@ -846,7 +846,7 @@ def handleCompletionInfo(self, completionsResp):
             if rawCompletions:
                 for rawCompletion in rawCompletions:
                     name = rawCompletion.name
-                    completion = (name + "\t" + rawCompletion.kind, name)
+                    completion = (name + "\t" + rawCompletion.kind, name.replace("$", "\$"))
```
Comment (DanielRosenwasser) [inline TypeScript.py]: Use `"\\$"`.

**Comment by DanielRosenwasser** [comment]
Not sure what this PR is supposed to do. Can you please clarify?

**Comment by billti** [comment]
Agree with Daniel, can you describe what is currently broken?  Is this related to $ potentially being interpreted as a snippet marker and thus needing to be escaped?  Please open an issue that describes the problem, and reference the relevant issue from any pull requests.

**Comment by DanielRosenwasser** [comment]
Context from [the link](http://www.sublimetext.com/forum/viewtopic.php?f=2&p=37311) given in #125:

> In snippets (rather than completions-files) dollar-signs need to be double-escaped \\ - but I should mention/apologise that I haven't proven my solution

**Comment by DanielRosenwasser** [comment]
:+1: I just pulled this down and it looks good; `replace` actually doesn't allocate a new string unless `$` occurs in the original name, so at least we avoid that overhead.

Any thoughts @billti?

**Comment by zhengbli** [comment]
I tried this solution with the dollar sign in different positions, and it all worked fine. :+1: 
Although if I start typing after the dollar sign, the completion list is still not right. However that should be a different issue, I'll file it elsewhere.

**Pull request merged**